frontend/src/metabase/visualizations/visualizations/Heading/Heading.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useMemo, MouseEvent } from "react";
+
+import cx from "classnames";
+import { t } from "ttag";
+
+import {
+  InputContainer,
+  HeadingContent,
+  HeadingContainer,
+  TextInput,
+} from "./Heading.styled";
+
+interface HeadingProps {
+  isEditing: boolean;
+  onUpdateVisualizationSettings: ({ text }: { text: string }) => void;
+  dashcard: { justAdded?: boolean };
```

Comment:
You can use `BaseDashboardOrderedCard` from `frontend/src/metabase-types/api/dashboard.ts`. And for settings on the line below we have `VisualizationSettings` from `frontend/src/metabase-types/api/card.ts`